### PR TITLE
Fix mail sending and add error logger

### PR DESCRIPTION
**Pull request by Michal-Dziedzinski**
### Short description and why it's useful

Fix mail sending and added error logger

### Screenshots of visual changes before/after (if there are any)

Before changes:
Zrzut ekranu (17)

After changes:
Zrzut ekranu (18)

### Which environment this relates to
Check your case. In case of any doubts please read about [Release Cycle](https://docs.vuestorefront.io/guide/basics/release-cycle.html)

- [x] Test version (https://test.storefrontcloud.io) - this is a new feature or improvement for Vue Storefront. I've created branch from `develop` branch and want to merge it back to `develop`
- [ ] RC version (https://next.storefrontcloud.io) - this is a stabilisation fix for Release Candidate of Vue Storefront. I've created branch from `release` branch and want to merge it back to `release`
- [ ] Stable version (https://demo.storefrontcloud.io) - this is an important fix for current stable version. I've created branch from `hotfix` or `master` branch and want to merge it back to `hotfix`

### Upgrade Notes and Changelog

- [x] No upgrade steps required (100% backward compatibility and no breaking changes)
- [x] I've updated the [Upgrade notes](https://github.com/DivanteLtd/vue-storefront/blob/develop/doc/Upgrade%20notes.md) and [Changelog](https://github.com/DivanteLtd/vue-storefront/blob/develop/CHANGELOG.md) on how to port existing VS sites with this new feature

**IMPORTANT NOTICE** - Remember to update `CHANGELOG.md` with description of your change

### Contribution and currently important rules acceptance

- [x] I read and followed [contribution rules](https://github.com/DivanteLtd/vue-storefront/blob/master/CONTRIBUTING.md)
- [ ] I read the [TypeScript Action Plan](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/TypeScript%20Action%20Plan.md) and adjusted my PR according to it
- [ ] I read about [Vue Storefront Modules](https://github.com/DivanteLtd/vue-storefront/blob/master/doc/api-modules/about-modules.md) and I am aware that every new feature should be a module

### CODE REVIEW

File: core/modules/mailer/store/index.ts [hunk 1]
```diff
@@ -25,12 +26,18 @@ export const mailerStore: Module<any, any> = {
                 })
               })
                 .then(res => resolve(res))
-                .catch(() => reject())
```
Comment (pkarw) [inline core/modules/mailer/store/index.ts]: can we refactor it to async/await form? @patzick will buy you a beer for that ;p

Comment (Michal-Dziedzinski) [reply core/modules/mailer/store/index.ts]: Sure, everything for beer! :D 🍻

**Comment by pkarw** [comment]
Hi @Michal-Dziedzinski when can I expect the changes to be applied? :-) More or less ;)?

**Comment by pkarw** [comment]
Thanks for the changes ;)

**Comment by alinadivante** [comment]
`Email has successfully been sent` 👌

**Pull request merged**